test/request.spec.js

```diff
@@ -758,14 +758,13 @@ describe('AWS.Request', function() {
         s = request.createReadStream();
         s.on('error', function(e) {
           error = e;
-          expect(error).to.be["null"];
+          expect(error).to.not.be["null"];
         });
         s.on('data', function(c) {
           return data += c.toString();
         });
         return s.on('end', function() {
-          expect(error).to.be["null"];
-          expect(data).to.equal('FOOBARBAZQU');
+          expect(error).to.not.be["null"];
```

Comment:
I can't imagine that the data received up to `content-length` would be usable in most cases... you would end up with an unparsable partial XML or JSON document unless the service was returning a byte stream. I think the change in this PR is correct, but we should be on the lookout for issues where users were receiving partial byte streams and are now encountering stream errors.